api/turing/middleware/openapi_validation.go

```diff
@@ -61,19 +81,27 @@ func NewOpenAPIV2Validation(swaggerYamlFile string, options OpenAPIValidationOpt
 		v3Swagger.Servers = nil
 	}
 
-	return &OpenAPIValidation{swagger: v3Swagger, router: router, options: opts}, nil
+	return &OpenAPIValidation{routers: routers, options: opts}, nil
 }
 
 // Validate the request against the OpenAPI spec
 func (openapi *OpenAPIValidation) Validate(r *http.Request) error {
-	route, pathParams, _ := openapi.router.FindRoute(r.Method, r.URL)
-	input := &openapi3filter.RequestValidationInput{
-		Request:    r,
-		PathParams: pathParams,
-		Route:      route,
-		Options:    openapi.options,
+	var err error
+	for _, router := range openapi.routers {
+		route, pathParams, _ := router.FindRoute(r.Method, r.URL)
+		input := &openapi3filter.RequestValidationInput{
+			Request:    r,
+			PathParams: pathParams,
+			Route:      route,
+			Options:    openapi.options,
+		}
+		err := openapi3filter.ValidateRequest(context.Background(), input)
+		if err != errors.New("invalid route") || err == nil {
+			// Make sure that it is the correct error message, ignore invalid route error unless that is the only error
```

Comment:
Is it possible to skip validation if `router.FindRoute` emits nil (can't find the route in the given spec)? So in this case you don't need to check if error is `invalid route`

Reply:
Because this is in two routers, other validation error messages should be taken priority.

Scenario one, GET `/api/router-1-route`:
router 1: `some other error`
router 2: `invalid route`

This scenario will fail without the because `err != errors.New("invalid route")` it will reply `invalid route` instead of some other error.

Scenario two, GET `/api/router-2-route`:
router 1: `invalid route`
router 2: `some other error`

This scenario will pass because router 2 (end of the loop) is the last error and `some other error` will appear.

Invalid route will be the catch all situation. The best we can do to cut short the loop is when it returns another error / no error during the loop. Else we have to test all routers.

That said, line 98 should be an assignment operator rather than a short variable declaration operator. Will change this.

Reply:
I'd prefer more explicit way of detecting if the endpoint is not described in any of specs. Something like this:

```go
            for idx, router := range openapi.routers {
		route, pathParams, _ := router.FindRoute(r.Method, r.URL)
                if route == nil && idx == len(openapi.routers) - 1 {
                     // endpoint is not described
                     return fmt.Errorf("Route `%s %s` is not described in openapi spec",  r.Method, r.URL)
                }
		input := &openapi3filter.RequestValidationInput{
			Request:    r,
			PathParams: pathParams,
			Route:      route,
			Options:    openapi.options,
		}
		if err := openapi3filter.ValidateRequest(context.Background(), input); err != nil {
                    return err
                }
           }
       return nil
```

Reply:
There's a small error there because once there is a valid route, you should always return the error since we know now that that error is valid.
```go
if err := openapi3filter.ValidateRequest(context.Background(), input); err != nil {
		return err
}
```

Anyway here's one that is working that should address your concerns.
```go
// Validate the request against the OpenAPI spec
func (openapi *OpenAPIValidation) Validate(r *http.Request) error {
	for idx, router := range openapi.routers {
		route, pathParams, _ := router.FindRoute(r.Method, r.URL)
		if route == nil && idx == len(openapi.routers)-1 {
			// endpoint is not described
			return fmt.Errorf("Route `%s %s` is not described in openapi spec", r.Method, r.URL)
		}
		input := &openapi3filter.RequestValidationInput{
			Request:    r,
			PathParams: pathParams,
			Route:      route,
			Options:    openapi.options,
		}
		return openapi3filter.ValidateRequest(context.Background(), input)
	}
	// Will not reach this line.
	return nil
}
```

Reply:
yup, lgtm